buildSrc/src/main/kotlin/SqlDelightTmdbCacheDefinitionsGenerator.kt

```diff
@@ -0,0 +1,110 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+/**
+ * A Gradle Task to generate the SQLDelight definitions related to the Tmdb Cache.
+ */
+open class SqlDelightTmdbCacheDefinitionsGenerator : DefaultTask() {
+    /** Where to generate the .sq files */
+    private val sqldelightRoot = File(project.projectDir, "src/androidMain/sqldelight/tmdbCache/io/github/couchtracker/db/tmdbCache")
+
+    /** Common columns */
+    private val movieIdColumn = SqlColumn.int("tmdbId", "io.github.couchtracker.tmdb.TmdbMovieId")
+    private val languageColumn = SqlColumn.text("language", "io.github.couchtracker.tmdb.TmdbLanguage")
+
+    /** Which caches to create */
+    private val caches = listOf(
+        SqlTable(
+            name = "MovieDetailsCache",
+            key = listOf(movieIdColumn, languageColumn),
+            value = SqlColumn.text("details", "app.moviebase.tmdb.model.TmdbMovieDetail"),
+        ),
+        SqlTable(
+            name = "MovieReleaseDatesCache",
+            key = listOf(movieIdColumn),
+            value = SqlColumn.text("releaseDates", "kotlin.collections.List<app.moviebase.tmdb.model.TmdbReleaseDates>"),
+        ),
+    )
+
+    /** Which sql functions to generate */
+    private val sqlFunctions = listOf(
+        SqlFunction("get") { cache ->
+            append("SELECT ${cache.value.name}\n")
+            append("FROM ${cache.name}\n")
+            append("WHERE " + cache.key.joinToString(" AND ") { "${it.name} = ?" } + ";\n")
+        },
+        SqlFunction("put") { cache ->
+            append("INSERT INTO ${cache.name}\n")
+            append("VALUES (${cache.key.joinToString { ":${it.name}" }}, :${cache.value.name})\n")
+            append("ON CONFLICT DO UPDATE SET ${cache.value.name} = :${cache.value.name};\n")
+        },
+    )
+
+    init {
+        group = "io.github.couchtracker"
+        description = "Generates the SqlDelight definitions for local the Tmdb Cache"
+    }
+
+    private data class SqlColumn(
+        val name: String,
+        val sqlType: String,
+        val kotlinType: String,
```

Comment:
Can we use a KType here instead? It retains generic type info: https://pl.kotl.in/iDHs2HQpC (we may need to add a dependncy to kotlin-reflect). This allows to have more complile-time safeties.

It will also avoid the need for the nullable boolean.

Reply:
No, this module cannot depend on the main module, so we cannot have static references to the classes